mlir/lib/Dialect/Rock/Transforms/ConvToGemm.cpp

```diff
@@ -607,25 +634,41 @@ LogicalResult backwardWeightAtomicAdd(Conv2DBwdWeightOp op,
     firstWrap.unmerge({"n0", "n1"}, "ni",
                       {gemmKBlocks, convDims.n / gemmKBlocks});
     firstWrap.passThrough("ci");
-    firstWrap.pad({"hipad", "wipad"}, {"hi", "wi"},
-                  {leftPadH, rightPadH, leftPadW, rightPadW});
+    SmallVector<StringRef, 3> outs;
+    SmallVector<StringRef, 3> ins;
+    for (size_t i = 0; i < convDims.in.size(); i++) {
+      outs.push_back(b.getStringAttr(Twine(i) + "ipad"));
+      ins.push_back(b.getStringAttr(Twine(i) + "i"));
+    }
+    firstWrap.pad(outs, ins, pads);
 
     TransformMapAttr firstTransformAttr = firstTransform.get();
     Value firstTransformed =
         b.create<TransformOp>(loc, op.getInput(), firstTransformAttr);
 
     // The usual mapping of input space to dimensions such that filter elements
     // get multiplied by the right thing
-    llvm::StringMap<uint32_t> embedOutDims = expandNamesInPlace(
-        firstTransform, {{"hipad", {"y", "ho"}}, {"wipad", {"x", "wo"}}});
+    expansions.clear();
+    for (size_t i = 0; i < convDims.out.size(); i++) {
+      StringAttr key = b.getStringAttr(Twine(i) + "ipad");
+      StringAttr val1 = b.getStringAttr(Twine(i));
+      StringAttr val2 = b.getStringAttr(Twine(i) + "o");
+      expansions.insert({key, {val1, val2}});
```

Comment:
Huh, does `StringMap` not own its keys?

Reply:
Nope, it maps StringRefs to things.  I even witnessed strings changing out from under the StringMap.